testing/integration-tests/src/full_client/client/mod.rs

```diff
@@ -121,6 +121,9 @@ async fn transaction_validation() {
 
     wait_for_blocks(&api).await;
 
+    #[cfg(feature = "unstable-reconnecting-rpc-client")]
+    let _ctx = ctx.restart().await;
```

Comment:
I would rather create a separate test for unstable-reconnecting stuff than injecting it here.

Also this would remove all feature-gated stuff in this PR and just feature-gate the test itself

Reply:
removed all the feature gating stuff and moved this to a separate test